ui/components/component-library/checkbox/checkbox.tsx

```diff
@@ -52,7 +52,7 @@ export const Checkbox: CheckboxComponent = React.forwardRef(
           'mm-checkbox--disabled': Boolean(isDisabled),
         })}
         as="label"
-        display={Display.Flex}
+        display={Display.InlineFlex}
```

Comment:
Updates the hit area to just the content 

### Before

https://github.com/MetaMask/metamask-extension/assets/8112138/7b038ded-4aea-4d63-9c6b-8b3da599ee86

### After


https://github.com/MetaMask/metamask-extension/assets/8112138/236e7496-257d-406b-b4eb-d576a61e3951